src/client/Microsoft.Identity.Client/Instance/Authority.cs

```diff
@@ -18,29 +21,37 @@ internal abstract class Authority
             },
             StringComparer.OrdinalIgnoreCase);
 
-        protected Authority(IServiceBundle serviceBundle, AuthorityInfo authorityInfo)
+        protected Authority(AuthorityInfo authorityInfo)
         {
-            ServiceBundle = serviceBundle;
-            AuthorityInfo = authorityInfo;
+            if (authorityInfo==null)
+            {
+                throw new ArgumentNullException(nameof(authorityInfo));
+            }
+
+            // Don't reuse the same authority info, instead copy it
+            // to prevent objects updating each other's details
+            AuthorityInfo = new AuthorityInfo(authorityInfo);
         }
 
         public AuthorityInfo AuthorityInfo { get; }
 
-        protected IServiceBundle ServiceBundle { get; }
```

Comment:
Having the servicebundle here is confusing and violates SRP. In fact, a single Authority ctor needs this, to be able to validate that the authority from the request is similar to the authority from the config. Much simpler to modify the affected ctor than to pass ServiceBundle everywhere.